Crash in Edit

I attempted to 2-3 click Edit on a pattern with `,x/.../` and got

```
panic: runtime error: index out of range

goroutine 119 [running]:
main.cmdexec(0xc420041930, 0xc4200ae190, 0xc42021c000)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/ecmd.go:108 +0x7e4
main.editthread()
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/edit.go:149 +0x99
created by main.editcmd
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/edit.go:231 +0x1d6
```

--- a/ecmd.go
+++ b/ecmd.go
@@ -441,7 +441,6 @@ func s_cmd(t *Text, cp *Cmd) bool {
 					j := c - '0'
 					if sel[j].q1-sel[j].q0 > RBUFSIZE {
 						editerror("replacement string too long")
-						return false
 					}
 					t.file.b.Read(sel[j].q0, rbuf[:sel[j].q1-sel[j].q0])
 					for k := 0; k < sel[j].q1-sel[j].q0; k++ {
@@ -455,7 +454,6 @@ func s_cmd(t *Text, cp *Cmd) bool {
 			} else {
 				if sel[0].q1-sel[0].q0 > RBUFSIZE {
 					editerror("right hand side too long in substitution")
-					return false
 				}
 				t.file.b.Read(sel[0].q0, rbuf[:sel[0].q1-sel[0].q0])
 				for k := 0; k < sel[0].q1-sel[0].q0; k++ {
@@ -960,7 +958,6 @@ func nextmatch(f *File, r string, p int, sign int) {
 		sel = are.rxbexecute(f.curtext, p, NRange)
 		if len(sel) == 0 {
 			editerror("no match for regexp")
-			sel = RangeSet{Range{0, 0}}
 		}
 		if sel[0].q0 == sel[0].q1 && sel[0].q1 == p {
 			p--

--- a/edit.go
+++ b/edit.go
@@ -2,6 +2,7 @@ package main
 
 import (
 	"fmt"
+	"runtime"
 	"runtime/debug"
 )
 
@@ -123,10 +124,9 @@ func init() {
 }
 
 var (
-	cmdstartp      []rune
-	cmdp           int
-	editerrc       chan error
-	endeditthreadc chan struct{}
+	cmdstartp []rune
+	cmdp      int
+	editerrc  chan error
 
 	lastpat string
 	patset  bool
@@ -135,20 +135,13 @@ var (
 )
 
 func editthread() {
-	endeditthreadc = make(chan struct{})
-loop:
 	for {
-		select {
-		case <-endeditthreadc:
+		cmd := parsecmd(0)
+		if cmd == nil {
 			break
-		default:
-			cmd := parsecmd(0)
-			if cmd == nil {
-				break loop
-			}
-			if !cmdexec(curtext, cmd) {
-				break loop
-			}
+		}
+		if !cmdexec(curtext, cmd) {
+			break
 		}
 	}
 	editerrc <- nil
@@ -196,7 +189,7 @@ func editerror(format string, args ...interface{}) {
 	s := fmt.Errorf(format, args...)
 	row.AllWindows(allelogterm) /* truncate the edit logs */
 	editerrc <- s
-	close(endeditthreadc) // exit.
+	runtime.Goexit()
 }
 
 func editcmd(ct *Text, r []rune) {

